### PR TITLE
feat(rpc): remove chain param from individual protocol ids

### PR DESCRIPTION
**Pull request by louise-poole**

### CODE REVIEW

File: tycho-client/src/feed/component_tracker.rs [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 use std::collections::{HashMap, HashSet};
 use tracing::{debug, instrument, warn};
+
 use tycho_core::{
-    dto::{BlockChanges, Chain, ProtocolComponent, ProtocolComponentsRequestBody, ProtocolId},
+    dto::{BlockChanges, Chain, ProtocolComponent, ProtocolComponentsRequestBody},
     Bytes,
 };
 
+use crate::{rpc::RPCClient, RPCError};
```
Comment (zizou0x) [inline tycho-client/src/feed/component_tracker.rs:9]: Why so many imports changes, is it RustRover?

File: tycho-indexer/src/services/mod.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 //! This module contains Tycho web services implementation
-
+#![allow(deprecated)]
```
Comment (zizou0x) [inline tycho-indexer/src/services/mod.rs:3]: Please add a TODO to remove else it's gonna stay there forever

File: tycho-core/src/dto.rs [hunk 3]
```diff
@@ -1036,8 +1056,122 @@ pub struct ProtocolStateRequestBody {
 }
 
 impl ProtocolStateRequestBody {
-    pub fn id_filtered(ids: Vec<ProtocolId>) -> Self {
-        Self { protocol_ids: Some(ids), ..Default::default() }
+    pub fn id_filtered<I, T>(ids: I) -> Self
+    where
+        I: IntoIterator<Item = T>,
+        T: Into<String>,
```
Comment (zizou0x) [inline tycho-core/src/dto.rs:1062]: Would it work with just one generic here? 

I: IntoIterator<Item = Into<String>>

File: tycho-client/src/feed/component_tracker.rs (hunk 1 above)
Comment (louise-poole) [reply tycho-client/src/feed/component_tracker.rs:9]: No - it was me. When I refactored the indexer earlier a cleaned up a lot of imports and made them consistent. Couldn't resist sneaking in doing that here too.

File: tycho-core/src/dto.rs (hunk 3 above)
Comment (louise-poole) [reply tycho-core/src/dto.rs:1062]: I previously tried to sombine it into one generic and got lots of errors. This was the cleanest solution I could come up with.

File: tycho-client/src/feed/component_tracker.rs [hunk 4]
```diff
@@ -178,10 +180,10 @@ where
             .collect()
     }
 
-    pub fn get_tracked_component_ids(&self) -> Vec<ProtocolId> {
+    pub fn get_tracked_component_ids(&self) -> Vec<String> {
         self.components
             .keys()
-            .map(|k| ProtocolId { chain: self.chain, id: k.clone() })
+            .cloned()
```
Comment (tvinagre) [inline tycho-client/src/feed/component_tracker.rs:186]: Is `ProtocolId` used with chain + id in more places? If not, we should refactor it and keep it as a String Type

Comment (louise-poole) [reply tycho-client/src/feed/component_tracker.rs:186]: We can do that when we actually remove it. For now it needs to remain as is for backwards compatibility.

**Pull request merged**